src/main/kotlin/com/wgslfuzz/core/Resolver.kt

```diff
@@ -476,6 +496,33 @@ private fun resolveAstNode(
             if (type.isAbstract()) {
                 type = defaultConcretizationOf(type)
             }
+
+            if (node.accessMode != null && node.addressSpace == null) {
+                throw IllegalArgumentException(
+                    "If an access mode is specified for a variable an address space must also be specified.",
+                )
+            } else if (node.addressSpace in
+                setOf(
+                    AddressSpace.PRIVATE,
+                    AddressSpace.STORAGE,
+                    AddressSpace.UNIFORM,
+                    AddressSpace.WORKGROUP,
+                    AddressSpace.HANDLE,
+                )
+            ) {
+                throw IllegalArgumentException(
+                    "The address spaces private, storage, uniform, workgroup and handle can only be used for module scope variables",
+                )
+            } else if (node.accessMode != null) {
+                throw IllegalArgumentException(
+                    "The access mode must not be specified unless a variable is in the storage address space.",
+                )
+            }
+
+            val addressSpace = node.addressSpace ?: AddressSpace.FUNCTION
```

Comment:
I guess this is fine - but due to the above you could simply make it be AddressSpace.FUNCTION, maybe with an assertion right before that this is what it is if it's not null.